Fix/custom provider

cds-amal opened the pull request:
This PR fixes #14. 

# How to test

**Note you will need 2 terminals**

## First terminal: 
- build out monorepo from the root project and start the ganache server
`   yarn && yarn ganache`

## Second terminal:
- deploy test contracts and start the test app
```
  yarn ganache:deploy
  cd ui-tests/custom-provider
  yarn serve:ui
```

## Verify 
Use the browser of your choice with metamask installed, visit [the test app](htpt://localhost:3003) and change SimpleStorage's StoredValue. The test is successful if you're able to change the value without MetaMask showing up.

cooper-sloan commented:
LGTM. Very useful.

matthewlilley commented:
Until MetaMask is updated, this seems to be a good solution. Can keep the state updated through https://infura.io/, but still send transactions through MetaMask provider.

keithpops commented:
Was just about to open a similar pull request. Thanks for this!

ccashwell commented:
Why is this not implemented yet?

james1465 commented:
@ccashwell Looking it the commits on the main repository, I don't think this is needed anymore. The author has implemented an alternative solution using polling so that all calls can go through metamask.

ccashwell commented:
Thanks @james1465, would be awesome if they'd update the community too... all these open issues and PRs make the library look like it's nowhere near ready for use.

DiscRiskandBisque commented:
First, let me apologize for the lack of attention this PR received. Lack of non-websocket provider (and therefore lack of MetaMask) support was not adequately communicated when Drizzle launched.

Today, Drizzle supports MetaMask via a fallback to their very own `web3-block-tracker` package (see (release v1.1.4)[https://github.com/trufflesuite/drizzle/releases/tag/1.1.4]). Release notes for v1.1 are also relevant as it was the first time MetaMask support rolled out.

Hope you'll all give it another shot if you haven't already! :)

cds-amal committed:
test: update web3 tests

cds-amal committed:
test: use explicit web3Options variable name

cds-amal committed:
fix: handle customProvider option correctly

cds-amal committed:
test: add customProvider test app

cds-amal committed:
build: enable CRA SKIP_PREFLIGHT_CHECK

cds-amal committed:
build: scope test to ui (test:ui)

cds-amal committed:
refactor(web3saga): custom provider is an object

cds-amal committed:
test(custom-provider): rename test app

cds-amal committed:
test(custom-provider): test custom-provider

Test modifications:
  - test resolved web3 is the custom-provider sent to web3Saga
  - remove test to verify networkId of custom-provider.

adrianmcli commented on packages/store/test/web3.test.js:
```diff
@@ -17,30 +17,17 @@ describe('Resolving Web3', () => {
-      hasWeb3Shape(resolvedWeb3)
-    })
-
-    test('get network ID', async () => {
```
@cds-amal just wondering why this test was removed.

cds-amal replied:
Checking the network id is an arbitrary test for a custom provider.
